sdk/src/Core/RuntimeDependencyRegistry.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Amazon.Runtime;
+using Amazon.Runtime.Internal;
+using AWSSDK.Runtime.Internal.Util;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Amazon
+{
+    /// <summary>
+    /// Their are some features of the AWS .NET SDK that at runtime load dependencies from assemblies. The most common
+    /// example is using credential profiles that require the service client from the AWSSDK.SecurityToken package.
+    /// Normally these runtime dependencies are resolved by using reflection to load the assembly and the type,
+    /// In Native AOT with trimming turned on that will not worked because the dependency would be trimmed out.
+    /// 
+    /// The RuntimeDependencyRegistry is used to work around this limitation by allowing users to explicitly register
+    /// the runtime dependency instance into the SDK removing any use of reflection.
+    /// </summary>
+    public class RuntimeDependencyRegistry : IDisposable
```

Comment:
Does this registry have a flaw in that there is only one static instance of the object of that type for the entire app? So the service instance you register would be locked into whatever region/settings of the one instance with no way to stop conflicts.